src/Console/Command/ListFilesCommand.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Command;
+
+use PhpCsFixer\Config;
+use PhpCsFixer\ConfigInterface;
+use PhpCsFixer\Console\ConfigurationResolver;
+use PhpCsFixer\ToolInfoInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Terminal;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * @internal
+ */
+final class ListFilesCommand extends Command
+{
+    protected static $defaultName = 'list-files';
+
+
+    /**
+     * @var ConfigInterface
+     */
+    private $defaultConfig;
+
+    /**
+     * @var ToolInfoInterface
+     */
+    private $toolInfo;
+
+    public function __construct(ToolInfoInterface $toolInfo)
+    {
+        parent::__construct();
+
+        $this->defaultConfig = new Config();
+        $this->toolInfo = $toolInfo;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Override here to only generate the help copy when used.
+     */
+    public function getHelp()
+    {
+        return HelpCommand::getHelpCopy();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDefinition(
```

Comment:
maybe we should have a switch to filter-out files that do not require a fixing (based on cache) ?

Reply:
👍 

Sounds good!

Reply:
I don't know the internals of php-cs enough to implement the mentioned switch. could you give me a hint of what you are talking about, or whether there is another code-site where I can get inspiration from?

Reply:
basically if the file has no changes, we do not need to fix it, thus we may not want to list it

you may want to take a look here:
- https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/4874565ef708da70481a13a21d0c28b39ef53008/src/Runner/Runner.php#L121
- https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/4874565ef708da70481a13a21d0c28b39ef53008/src/Runner/FileFilterIterator.php

Reply:
for the use-case I foresee (see the referenced issue), the cache will be empty in most situations.
IMO we could add support for using the cache later, in case we get use-cases which benefit from it.

since this new option would add additional complexity and takes time, I would initially add the feature without a cache option - so we can get the basic feature into the tool earlier.